### PR TITLE
Fix: integration test 025 is not isolated

### PR DESCRIPTION
**Pull request by sagilaufer1992**
### Issue & Steps to Reproduce / Feature Request
[//]: <> (choose “resolves” “fixes” or “closes” and put link for the issue)

test `025_notifications` uses `all_notifications`, and it fetches data from other tests, such as `017_notification`, that may become unavailable during test 025's run.

### Solution

make the test rely solely on its own data

### QA done

ran the test locally to make sure it works on its own. deliberately failing it is much harder, didn't happen sporadically (that we caught) until a few days prior to the PR

### CODE REVIEW

File: tests/integration/025_notifications/main.tf [hunk 1]
```diff
@@ -18,9 +18,14 @@ resource "env0_notification" "test_notification_2" {
   value = "https://someurl2.com"
 }
 
-data "env0_notifications" "all_notifications" {}
+data "env0_notifications" "all_notifications" {
+  depends_on = [env0_notification.test_notification_1, env0_notification.test_notification_2]
+}
+
+data "env0_notification" "test_notification_1" {
```
Comment (avnerenv0) [inline tests/integration/025_notifications/main.tf:25]: why not just use 
```
data "env0_notification" "my_notification" {
  name = locals.test_notification_1_name
}
```
here?

Comment (sagilaufer1992) [reply tests/integration/025_notifications/main.tf:25]: we wish to see that the `all_notifications` works as expected too, so I suggest we utilize its data, so if it got broken for some reason the test will fail

Comment (chpl) [reply tests/integration/025_notifications/main.tf:25]: agree with @avnerenv0 .
can be replaced with post condition on the `all_notifications`

Comment (sagilaufer1992) [reply tests/integration/025_notifications/main.tf:29]: [0c08f76](https://github.com/env0/terraform-provider-env0/pull/735/commits/0c08f76aa77d2f10d0a9144aa5caf8222804f19b)

used the resource name, and used output for the `all_notifications` verification, that it got one of the notifications

**Review by TomerHeber — APPROVED** [review]

**Pull request merged**